daemon/container_operations.go

```diff
@@ -118,6 +118,24 @@ func (daemon *Daemon) buildSandboxOptions(container *container.Container) ([]lib
 		sboxOptions = append(sboxOptions, libnetwork.OptionExtraHost(parts[0], parts[1]))
 	}
 
+	// Add a DNS Entry that maps host.docker.internal to default network bridge's (docker0) IP
+	if daemon.configStore.DNSResolveDockerHost {
+		if n, err := daemon.netController.NetworkByName(defaultNetName); err == nil {
+			v4Info, v6Info := n.Info().IpamInfo()
+			var gateway string
+			if len(v4Info) > 0 {
+				gateway = v4Info[0].Gateway.IP.String()
+
+			} else if len(v6Info) > 0 {
+				gateway = v6Info[0].Gateway.IP.String()
+			}
+			if len(gateway) > 0 {
+				sboxOptions = append(sboxOptions, libnetwork.OptionExtraHost(network.DockerHostInternalDNSName, gateway))
+				sboxOptions = append(sboxOptions, libnetwork.OptionExtraHost(network.DockerGatewayInternalDNSName, gateway))
```

Comment:
Actually; I see you mention the reverse;

> This PR allows containers to connect to Linux hosts by setting a Daemon flag called dns-resolve-docker-host which adds a host.docker.internal entry in /etc/hosts and maps it to docker0's IP

`/etc/hosts` is only used on the default (bridge) network, but isn't used for DNS resolution on other networks (well, I guess it's used, but it would be different from resolution of other entries we set in the embedded DNS)?


edit: hm.. so we actually _do_ use it for the custom hosts (`--add-host`), so this would be consistent. Would it be better to use the embedded DNS?

```bash
docker network create foo 

docker run --rm --network=foo --add-host foo.example.com:123.123.123.1 busybox cat /etc/hosts

127.0.0.1	localhost
::1	localhost ip6-localhost ip6-loopback
fe00::0	ip6-localnet
ff00::0	ip6-mcastprefix
ff02::1	ip6-allnodes
ff02::2	ip6-allrouters
123.123.123.1	foo.example.com
172.19.0.2	de9b9822b00f
```

Reply:
What happens if this is, for example, a macvlan interface? Wouldn't the gateway be different (ie. not necessarily the "host" ip) in such a case?